### PR TITLE
fix: generate proper number of segments for multiperiod content that uses segment template

### PR DESCRIPTION
**Pull request by gesinger**

### CODE REVIEW

File: test/inheritAttributes.test.js [hunk 1]
```diff
@@ -1370,7 +1374,7 @@ QUnit.test(
     const actual = toPlaylists(inheritAttributes(stringToMpdXml(`
     <MPD mediaPresentationDuration= "PT30S"  >
       <BaseURL>https://www.example.com/base</BaseURL>
-      <Period duration= "PT0H4M40.414S" >
+      <Period>
```
Comment (gesinger) [inline test/inheritAttributes.test.js:1377]: Here and below, the duration of the period exceeded the `mediaPresentationDuration`, so removed the period duration to fix the manifests (they were breaking the tests after this change).

Comment (brandonocasey) [reply test/inheritAttributes.test.js:1377]: Do we still have tests with period duration?

**Review by gkatsev — APPROVED** [review]

File: test/inheritAttributes.test.js (hunk 1 above)
Comment (gesinger) [reply test/inheritAttributes.test.js:1377]: In inheritAttributes I just added one recently! https://github.com/videojs/mpd-parser/blob/main/test/inheritAttributes.test.js#L675

And for segment generation I added https://github.com/videojs/mpd-parser/pull/138/files#diff-fcc9322a475e1685d979ab8e7c8da4736e1653a4613e5459ded4b47e8440edc8R1516

For end-to-end I don't believe we have any right now. I can add one.

**Review by brandonocasey — APPROVED** [review]
Just the one question about manifests with period duration.

File: test/inheritAttributes.test.js (hunk 1 above)
Comment (brandonocasey) [reply test/inheritAttributes.test.js:1377]: Yeah might be good to add one.

Comment (gesinger) [reply test/inheritAttributes.test.js:1377]: 👍 added

**Review by brandonocasey — APPROVED** [review]

**Pull request merged**